x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googleaistudio/GoogleAiStudioService.java

```diff
@@ -213,6 +224,13 @@ protected void doChunkedInfer(
         TimeValue timeout,
         ActionListener<List<ChunkedInferenceServiceResults>> listener
     ) {
-        throw new UnsupportedOperationException("Chunked inference not supported yet for Google AI Studio");
+        GoogleAiStudioModel googleAiStudioModel = (GoogleAiStudioModel) model;
+        var actionCreator = new GoogleAiStudioActionCreator(getSender(), getServiceComponents());
+
+        var batchedRequests = new EmbeddingRequestChunker(input, EMBEDDING_MAX_BATCH_SIZE).batchRequestsWithListeners(listener);
+        for (var request : batchedRequests) {
+            var action = googleAiStudioModel.accept(actionCreator, taskSettings, inputType);
+            action.execute(new DocumentsOnlyInput(request.batch().inputs()), timeout, request.listener());
+        }
```

Comment:
We also should added an implementation to determine the dimensions in the response.

Here's an example: https://github.com/elastic/elasticsearch/blob/a304366f42da23729c1be4328aa8c8818d3e526e/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/CohereService.java#L265-L293

Do you know what the similarity should default to? It depends on the model so might be worth poking around in the google gemini docs.

Reply:
Adjusted with [Implement checkModelConfig in GoogleAiStudioService](https://github.com/elastic/elasticsearch/pull/109243/commits/a1bc01376b1996105d7d2537d4379b418437f284). 

> Do you know what the similarity should default to?

Didn't found any specific recommendations, but they use dot product in their examples, so setting the default to that.